storage.go

```diff
@@ -27,11 +27,16 @@ func (s *Storage) createDir(ctx context.Context, path string, opt pairStorageCre
 	path = s.getAbsPath(path)
 	pathUnits := strings.Split(path, "/")
 	parentsId := "root"
+	f := s.cacheDir()
+
 	for _, v := range pathUnits {
 		parentsId, err = s.mkDir(ctx, parentsId, v)
 		if err != nil {
 			return nil, err
 		}
+
+		f(v, parentsId)
+
```

Comment:
How about calculating the path in the loop:

```suggestion
	currentPath := s.workDir

	for _, v := range pathUnits {
		parentsId, err = s.mkDir(ctx, parentsId, v)
		if err != nil {
			return nil, err
		}
                
                currentPath += "/" + v
		s.setCache(currentPath, parentsId)
```

Reply:
Yes, I do agree with this, just like what we do before this commit. Maybe I just misunderstood you when you say:

> Maybe we can extract them as a helper function?

Reply:
>currentPath := s.workDir

But this may not work. Because this will make our path like `/a`, `/a/b`, `/a/b/c`.  But we parse our path `a/b/c` like `a`, `b`, `c`, so we can never successfully find its cache.

We should do this:
```
currentPath := ""
for _, v := range pathUnits {
	parentsId, err = s.mkDir(ctx, parentsId, v)
	if err != nil {
		return nil, err
	}

	if currentPath == "" {
		currentPath = v
	}else {
		currentPath += "/" + v

	}

	s.setCache(currentPath, parentsId)

```